File: challenger-backend/src/main/java/com/kameo/challenger/config/ServerConfig.java
package com.kameo.challenger.config;

public class ServerConfig {
    private static String confirmEmailInvitationPattern = "http://blablah/uapi/confirm/{hash}";
    private static boolean crossDomain = true;

    public static String getConfirmEmailInvitationPattern(String uid) {
        return confirmEmailInvitationPattern.replace("{hash}", uid);
    }

    public static boolean isCrossDomain() {
        return crossDomain;
    }

    public static void setCrossDomain(boolean crossDomain) {
        ServerConfig.crossDomain = crossDomain;
    }
}

File: challenger-backend/src/main/java/com/kameo/challenger/config/ServerConfig.kt
package com.kameo.challenger.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component


@Component
@Scope(ConfigurableBeanFactory.SCOPE_SINGLETON)
open class ServerConfig {

    @Value("\${serverUrl}")
    lateinit var serverUrl: String


    var isCrossDomain = true
    val maxEventsSize:Int=500


    fun getConfirmEmailInvitationPattern(uid: String): String {
        return "$serverUrl#action={hash}".replace("{hash}", uid)
    }

    fun getLoginLink(): String {
        return serverUrl
    }

    companion object {

        /**
         * Web services base path
         */
        @JvmField val vsPATH="/api"

        /**
         * Web services subPath
         */
        const val restPath=""
    }

}
